src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.cs

```diff
@@ -2205,19 +2205,22 @@ internal Node RemoveAll(Predicate<T> match)
                 Contract.Ensures(Contract.Result<Node>() != null);
 
                 var result = this;
-                int index = 0;
-                foreach (var item in this)
-                {
+                var index = 0;
+                var count = this.Count;
+                var removed = 0;
+                for(;;)
```

Comment:
OK... I've seen quite a mix, but I'll switch to while (true).  Actually, can move it to a while loop with a condition given the break is the first thing in the body.  Will fix.